internal/cmd/common.go

```diff
@@ -59,20 +62,42 @@ type MonitorLoopParams struct {
 	IdentitySearchFn         func(ctx context.Context, config *notifications.IdentityMonitorConfiguration, monitoredValues identity.MonitoredValues) ([]identity.MonitoredIdentity, []identity.FailedLogEntry, error)
 }
 
+var (
+	once = flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
+)
+
+func handleError(msg string, err error) {
+	errWrap := errors.Join(errors.New(msg), err)
+	fmt.Fprint(os.Stderr, errWrap, "\n")
+
+	if !*once {
+		errStr := errWrap.Error()
+		// // These specific messages are expected in normal operation and are not treated as consistency check failures.
+		// // Therefore, they are excluded from Prometheus failure metrics.
+		if strings.Contains(errStr, "consistency proofs can not be computed starting from an empty log") ||
+			strings.Contains(errStr, "no start index set and no log checkpoint") {
+			return
+		}
+		server.IncLogIndexVerificationFailure()
+	} else {
+		os.Exit(1)
+	}
+}
+
 type Checkpoint interface{}
 type LogInfo interface{}
 
 // ParseMonitorFlags parses command-line flags and returns a MonitorFlags struct
 func ParseMonitorFlags(defaultServerURL, defaultTUFRepository string, baseUserAgentName string) *MonitorFlags {
 	configFilePath := flag.String("config-file", "", "path to yaml configuration file containing identity monitor settings")
 	configYamlInput := flag.String("config", "", "string with YAML configuration containing identity monitor settings")
-	once := flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
```

Comment:
you can keep this here instead of https://github.com/sigstore/rekor-monitor/pull/665/files#diff-83a2daadb61b967d7cd2c064c50ee421a2b730cc93a98c1bad36089f863d446bR65-R67 and use `params.Once` in the `MonitorLoop` function